pkg/data/management/role_data.go

```diff
@@ -54,6 +54,8 @@ func addRoles(wrangler *wrangler.Context, management *config.ManagementContext)
 		addRule().apiGroups("management.cattle.io").resources("cisconfigs").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("cisbenchmarkversions").verbs("get", "list", "watch").
 		addRule().apiGroups("rke-machine-config.cattle.io").resources("*").verbs("create").
+		// this permission is required for rke2 `force-delete` to work on nodes from the UI.
```

Comment:
Don't want to mention UI specifically as it's not specific to UI
```suggestion
		// this permission is required for rke2 `force-delete` to work as it's done by adding annotation to machine object
```

Reply:
It looks like `cluster-owner` actually already has a wildcard permission for `rke-machine.cattle.io`:  https://github.com/rancher/rancher/blob/93dbfff1a2d021df83407fe648c4c3be7fca7a7a/pkg/data/management/role_data.go#L153

but upon doing the same testing after granting a standard user `cluster-owner` the issue persists. So it might not be something related to the role.